aten/src/ATen/native/cuda/SegmentReduce.cu

```diff
@@ -172,6 +191,12 @@ __global__ static void segment_reduce_backward_kernel(
       int64_t starting_index = (j * stride_count) + lane_id;
       grad_input_data[starting_index] = grad_val;
     }
+  } else if (reduction == SegmentReductionType::SUM) {
```

Comment:
it's possible that for multi-d sum, backward can be done more efficiently with index_select, but it requires benchmarks (no changes needed, note for the future)